spot_rl_experiments/spot_rl/envs/gaze_env.py

```diff
@@ -3,103 +3,135 @@
+        self.policy.reset()
+
+        self.gaze_env = SpotGazeEnv(config, spot)
+        self.gaze_env.power_robot()
+
+    def gaze(self, target_object_list):
+        gaze_success_list = []
+        print(f"Target object list : {target_object_list}")
+        for target_object in target_object_list:
+            observations = self.gaze_env.reset(target_obj_name=target_object)
+            done = False
+            start_time = time.time()
+            self.gaze_env.say(f"Gaze at target object - {target_object}")
+            while not done:
+                action = self.policy.act(observations)
+                observations, _, done, _ = self.gaze_env.step(arm_action=action)
+            self.gaze_env.say("Gaze finished")
+
+            gaze_success_list.append(
+                {
+                    "target_object": target_object,
+                    "time_taken": time.time() - start_time,
+                    "success": done,
+                }
+            )
+        return gaze_success_list
+
+    def get_gripper_images(self):
+        # CHECK IS THIS IS USEFUL
+        arm_depth, arm_depth_bbox = self.gaze_env.get_gripper_images()
+        return arm_depth, arm_depth_bbox
+
+    def shutdown(self, should_dock=False) -> None:
+        """
+        Stops the robot and docks it if should_dock is True else sits the robot down
+
+        Args:
+            should_dock: bool indicating whether to dock the robot or not
+        """
+        try:
+            if should_dock:
+                self.gaze_env.say("Executing automatic docking")
+                dock_start_time = time.time()
+                while time.time() - dock_start_time < 2:
+                    try:
+                        self.spot.dock(dock_id=DOCK_ID, home_robot=True)
+                    except Exception:
+                        print("Dock not found... trying again")
+                        time.sleep(0.1)
             else:
-                cmd_id = spot.set_base_position(
-                    x_pos=goal_x, y_pos=goal_y, yaw=goal_heading, end_time=100
-                )
-
-        x, y, yaw = spot.get_xy_yaw()
-        dist = np.linalg.norm(np.array([x, y]) - np.array([goal_x, goal_y]))
-        heading_diff = abs(wrap_heading(goal_heading - yaw))
-        success = dist < 0.3 and heading_diff < np.deg2rad(5)
-
-    return traj
-
-
-def close_enough(pos1, pos2):
-    dist = np.linalg.norm(np.array([pos1[0] - pos2[0], pos1[1] - pos2[1]]))
-    theta = abs(wrap_heading(pos1[2] - pos2[2]))
-    return dist < 0.1 and theta < np.deg2rad(2)
+                self.gaze_env.say("Will sit down here")
+                self.spot.sit()
+        finally:
+            self.spot.power_off()
 
 
 class SpotGazeEnv(SpotBaseEnv):
-    def reset(self, target_obj_id=None, *args, **kwargs):
+    def __init__(self, config, spot):
+        super().__init__(config, spot)
+        self.target_obj_name = self.config.TARGET_OBJ_NAME
+
```

Comment:
The init was needed here so that if one initializes SpotGazeEnv with a set of configs, then SpotBaseEnv also gets initialized with the same set of configs.